Lib/pathlib/_abc.py

```diff
@@ -640,23 +640,22 @@ def write_text(self, data, encoding=None, errors=None, newline=None):
         with self.open(mode='w', encoding=encoding, errors=errors, newline=newline) as f:
             return f.write(data)
 
-    def scandir(self):
+    def _scandir(self):
         """Yield os.DirEntry objects of the directory contents.
 
         The children are yielded in arbitrary order, and the
         special entries '.' and '..' are not included.
         """
-        raise UnsupportedOperation(self._unsupported_msg('scandir()'))
+        import contextlib
+        return contextlib.nullcontext(self.iterdir())
```

Comment:
This would need to construct `os.DirEntry` objects in order to be correct, yes?

Reply:
It needs to construct objects with a `name` attribute, and `is_dir()` and `is_symlink()` methods, and so `PathBase` objects are fine. I'm planning to add a protocol for this at some point.

Reply:
Perhaps it would be better if I added that protocol now, or at least improved the docstring?

Reply:
At the very least improve the docstring, yeah. We shouldn't mention `os` at all.